---
title: "Data collection"
format: html
---

## Data sets

Data sets were newly curated for the current study and originates from both mouse and human embryos. Murine samples were collected at E9.5 (n = 3), E11.5 (n = 2), and E13.5 (n = 2) from pregnant females. The collection and use of human embryos, which obtained from women who voluntarily terminated their pregnancies in this study were approved by the Clinical Research Ethics Committee of the First Affiliated Hospital of Zhejiang University. Based on size and anatomical characteristics, human embryos were staged as CS12, CS14, CS16, and CS18. And only one sample was representative for each stage.

After frozen sectioning, serial slides from target embryos were collected and one section stained with hematoxylin and eosin (H&E). Slides approximating the midsagittal plane were mounted onto 10 × Genomics Visium array slides (6.5 × 6.5 mm or 11 × 11 mm area) for spatially resolved transcriptomics with Visium platform following the manufacturer's instructions outlined in the user guides. Another 2 adjacent sections were mounted onto Superfrost Plus slides (Citotest) for AFADESI-MSI, with vacuum-dried at first. Raw data were converted to the standard ‘.imzML’ format using the imzMLConverter software and ion identification was conducted with pySM framework. Spatial transcriptomics and metabolomics data were gained from Ouyi Biological Company. Tissue annotations were manually performed using graphical user interface of Loupe Browser (v8) according to morphological structure from H&E and transcript features of unsupervised clustering. 

Additionally, a single-nucleus RNA sequencing dataset (GSE186069) was downloaded from GEO database with its associated annotations, to characterize cell composition within embryonic sample spots. 
